### PR TITLE
Remove apiv3 charmworld code

### PR DESCRIPTION
**Pull request by hatched**
This is the final branch in the removal of the apiv3 charmworld code. The GUI now relies exclusively on the new apiv4.

### CODE REVIEW

**Comment by hatched** [comment]
#### To QA

Depp Exploratory QA

File: test/index.html [hunk 1]
```diff
@@ -94,10 +94,9 @@
   <script src="test_bundle_details_view.js"></script>
   <script src="test_bundle_module.js"></script>
   <script src="test_bundle_import_helpers.js"></script>
-  <script src="test_charmworld.js"></script>
   <script src="test_token_container.js"></script>
   <script src="test_token.js"></script>
-  <script src="test_token_drag_and_drop.js"></script>
+  <script src="test_token_drag_and_drop.js"></script>-->
```
Comment (makyo) [inline test/index.html]: Extra close-comment?

**Comment by makyo** [comment]
QA problem: Clicking the link to view the charm on a deployed service gets a 404: `GET https://api.jujucharms.com/charmstore/v4/cs:precise/quantum-gateway-24/meta…ude=charm-config&include=manifest&include=charm-related&include=extra-info 404 (Not Found)` (note the "cs:") in there.  Other than that, looking good so far.  Going to do a bit of live env testing, too.

**Comment by jujugui** [comment]
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/juju-gui/2432/
Test PASSed.

File: test/index.html (hunk 1 above)
Comment (hatched) [inline test/index.html]: Oops good find, I even looked through it a few times for left overs hah!

**Comment by hatched** [comment]
Good finds, I've updated to fix that bug.

**Comment by makyo** [comment]
:+1: QA okay - everything looks great now.  Thanks for getting this all switched over!

**Comment by jujugui** [comment]
Test FAILed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/juju-gui/2433/
Test FAILed.

**Comment by kadams54** [comment]
Two QA issues noticed while testing in a sandbox:

* Bugs link in the details panel doesn't seem to work.
* Social sharing icons aren't clickable… maybe they're sorta "hidden" behind other layers?

I'm not sure these are connected with the v4 work, so we may just need separate cards for them.

**Comment by kadams54** [comment]
QA notes:

* Deployed Wordpress via "Add to my canvas" button from the "Popular" list.
* Deployed MySQL via drag-n-drop.
* Search for Apache. Deployed via drag-n-drop.
* Clicked the "Home" link.
* Searched for "mongo".
* Cleared search and submitted an empty string.
* Opened up detail panel for Ubuntu.
* Switched among all the tabs.
* Verified that clicking on another charm opened the detail panel in the same tab.
* Went to the code tab and verified that I could load/view a source file.
* Went to related tab and verified that clicking on a related charm loaded it into the details panel.
* Expanded change log in summary tab.
* Checked the source and bugs links.
* Checked the social sharing links.

**Comment by kadams54** [comment]
:+1: for the code.

**Comment by hatched** [comment]
Thanks for the reviews and qa's! :shipit:

**Comment by jujugui** [comment]
Status: merge request accepted. Url: http://ci.jujugui.org:8080/job/juju-gui-merge

**Comment by jujugui** [comment]
Build failed: Attempt to land pull request failed
        build url: http://ci.jujugui.org:8080/job/juju-gui-merge/853

**Comment by jujugui** [comment]
Status: merge request accepted. Url: http://ci.jujugui.org:8080/job/juju-gui-merge

**Pull request merged**